client/my-sites/earn/launchpad/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { CircularProgressBar } from '@automattic/components';
+import { useLaunchpad, LaunchpadNavigator } from '@automattic/data-stores';
+import { Launchpad, Task, setUpActionsForTasks } from '@automattic/launchpad';
+import { useDispatch } from '@wordpress/data';
+import { useTranslate } from 'i18n-calypso';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSelector } from 'calypso/state';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+const EarnLaunchpad = () => {
+	const translate = useTranslate();
+
+	const site = useSelector( ( state ) => getSelectedSite( state ) );
+
+	const {
+		data: { checklist },
+	} = useLaunchpad( site?.slug ?? null, 'earn' );
+
+	const { setActiveChecklist } = useDispatch( LaunchpadNavigator.store );
+
+	const numberOfSteps = checklist?.length || 0;
+	const completedSteps = ( checklist?.filter( ( task: Task ) => task.completed ) || [] ).length;
+	const tasklistCompleted = completedSteps >= numberOfSteps;
+
+	const tracksData = {
+		recordTracksEvent,
+		checklistSlug: 'earn',
+		tasklistCompleted,
+		launchpadContext: 'earn',
+	};
+
+	function shouldLoadLaunchpad() {
+		return numberOfSteps > 0;
+	}
+
+	const taskFilter = ( tasks: Task[] ) => {
```

Comment:
@edanzer I want to advise you that I'm creating a component to wrap the Launchpad and prevent the repetition of this code.

I also added the track events for viewing the tasks and task list.

https://github.com/Automattic/wp-calypso/pull/82612

Reply:
Thanks for the heads up!

Reply:
@valterlorran - Looks like that PR was merged. I rebased and tried removing the extra code in my PR around the taskFilter. But when I do, I still see a situation where nothing happens when clicking the links. Just FYI.